internal/disk/disk.go

```diff
@@ -18,6 +30,63 @@ const (
 	DefaultGrainBytes = uint64(1024 * 1024) // 1 MiB
 )
 
+// Entity is the base interface for all disk-related entities.
+type Entity interface {
+	// IsContainer indicates if the implementing type can
+	// contain any other entities.
+	IsContainer() bool
+
+	// Clone returns a deep copy of the entity.
+	Clone() Entity
+}
+
+// Container is the interface for entities that can contain other entities.
+// Together with the base Entity interface this allows to model a generic
+// entity tree of theoretically arbitrary depth and width.
+type Container interface {
+	Entity
+
+	// GetItemCount returns the number of actual child entities.
+	GetItemCount() uint
+
+	// GetChild returns the child entity at the given index.
+	GetChild(n uint) Entity
+}
+
+// Sizeable is implemented by entities that carry size information.
+type Sizeable interface {
+	// EnsureSize will resize the entity to the given size in case
+	// it is currently smaller. Returns if the size was changed.
+	EnsureSize(size uint64) bool
+
+	// GetSize returns the size of the entity in bytes.
+	GetSize() uint64
+}
+
+// A Mountable entity is an entity that can be mounted.
+type Mountable interface {
+
+	// GetMountPoint returns the path of the mount point.
+	GetMountpoint() string
+
+	// GetFSType returns the file system type, e.g. 'xfs'.
+	GetFSType() string
+
+	// GetFSSpec returns the file system spec information.
+	GetFSSpec() FSSpec
+
+	// GetFSTabOptions returns options for mounting the entity.
+	GetFSTabOptions() FSTabOptions
+}
+
+// A VolumeContainer is a container that is able to create new volumes.
+//
+// CreateVolume creates a new volume with the given mount point and
+// size and returns it.
+type VolumeContainer interface {
```

Comment:
As discussed, I found the concept of `Volume` a bit confusing here. Maybe more documentation could fix that, but renaming to `MountableContainer` and `CreateMountable()` would also work, and I think that would make it self explanatory.

Reply:
This is clearly a candidate for follow-up work. Marking as resolved.